packages/x-data-grid/src/utils/css/context.tsx

```diff
@@ -0,0 +1,50 @@
+import * as React from 'react';
+import { useTheme } from '@mui/material/styles';
+import { hash } from '@mui/x-internals/hash';
+import { transformMaterialUITheme } from './transformVariables';
+
+const CLASSNAME_PREFIX = 'MuiDataGridVariables';
+
+const CSSVariablesContext = React.createContext({
+  className: 'unset',
+  tag: <style href="/unset" />,
+});
```

Comment:
Similar to the `mui-next` prototype, this PR inserts an inline `style` tag to inject the classname for CSS variables.